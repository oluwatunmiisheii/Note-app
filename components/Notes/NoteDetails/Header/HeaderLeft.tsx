import { AntDesign } from "@expo/vector-icons";
import { useNavigation } from "@react-navigation/native";
import { TouchableOpacity } from "react-native";

export const HeaderLeft = () => {
  const navigation = useNavigation();
  return (
    <TouchableOpacity onPress={() => navigation.goBack()}>
      <AntDesign
        name="arrowleft"
        color="white"
        size={20}
        style={{ marginRight: 20 }}
      />
    </TouchableOpacity>
  );
};
