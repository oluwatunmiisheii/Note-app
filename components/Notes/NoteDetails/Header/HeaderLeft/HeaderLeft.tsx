import { AntDesign } from "@expo/vector-icons";
import { useNavigation } from "@react-navigation/native";
import { TouchableOpacity } from "react-native";

export const HeaderLeft = () => {
  const navigation = useNavigation();
  return (
    <TouchableOpacity
      onPress={() => navigation.goBack()}
      accessibilityLabel="go to previous page"
    >
      <AntDesign
        name="arrowleft"
        color="white"
        size={22}
        style={{ marginRight: 20 }}
      />
    </TouchableOpacity>
  );
};
