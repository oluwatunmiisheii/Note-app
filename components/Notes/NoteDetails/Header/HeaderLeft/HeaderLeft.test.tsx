import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react-native";
import { HeaderLeft } from "./HeaderLeft";

const create = () => {
  const tree = renderer.create(<HeaderLeft />).toJSON();
  return { tree };
};

const setup = () => {
  const utils = render(<HeaderLeft />);
  return { ...utils };
};

const mockGoBack = jest.fn();

jest.mock("@react-navigation/native", () => ({
  useNavigation: jest.fn(() => ({
    goBack: mockGoBack,
  })),
}));

describe(`<${HeaderLeft.name}>`, () => {
  test("renders correctly", () => {
    const { tree } = create();
    expect(tree).toMatchSnapshot();
  });

  test("should redirect to previous page on tap", async () => {
    setup();
    const trigger = screen.getByLabelText("go to previous page");

    fireEvent(trigger, "press");

    expect(mockGoBack).toBeCalledTimes(1);
  });
});
