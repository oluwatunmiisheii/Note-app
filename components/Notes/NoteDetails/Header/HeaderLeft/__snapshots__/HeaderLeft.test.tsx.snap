// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLeft> renders correctly 1`] = `
<View
  accessibilityLabel="go to previous page"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <AntIcon
    color="white"
    name="arrowleft"
    size={20}
    style={
      Object {
        "marginRight": 20,
      }
    }
  />
</View>
`;
